### PR TITLE
feat: Prepare for adding CDK docs

### PR DESCRIPTION
**Pull request by josephperrott**
#243

### CODE REVIEW

File: src/app/pages/component-sidenav/component-sidenav-can-load-guard.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import {Injectable} from '@angular/core';
+import {CanActivate, ActivatedRouteSnapshot, Router, RouterStateSnapshot} from '@angular/router';
+import {SECTIONS} from '../../shared/documentation-items/documentation-items';
+
+@Injectable()
+export class CanActivateComponentSidenav implements CanActivate {
```
Comment (jelbourn) [inline src/app/pages/component-sidenav/component-sidenav-can-load-guard.ts]: Add class description?

File: src/app/pages/component-sidenav/component-sidenav-can-load-guard.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  constructor(private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    let sectionFound = Object.keys(SECTIONS).find(
```
Comment (jelbourn) [inline src/app/pages/component-sidenav/component-sidenav-can-load-guard.ts]: Add comment explaining what this is doing? E.g. `Prevent navigation to a section if...`

File: src/app/pages/component-category-list/component-category-list.ts [hunk 3]
```diff
@@ -13,11 +15,16 @@ import {SvgViewerModule} from '../../shared/svg-viewer/svg-viewer';
...
+  params: Observable<Params>;
+
   constructor(public docItems: DocumentationItems,
-              public _componentPageTitle: ComponentPageTitle) {}
+              public _componentPageTitle: ComponentPageTitle,
+              private _route: ActivatedRoute) {}
 
   ngOnInit() {
-    this._componentPageTitle.title = 'Component Categories';
+    this.params = Observable.combineLatest(
```
Comment (jelbourn) [inline src/app/pages/component-category-list/component-category-list.ts]: Add comment here?

File: src/app/shared/documentation-items/documentation-items.ts [hunk 4]
```diff
@@ -10,96 +10,115 @@ export interface DocCategory {
...
-      {id: 'sort', name: 'Sort header', examples: ['sort-overview']},
-      {id: 'paginator', name: 'Paginator', examples: ['paginator-configurable']},
-    ]
-  }
-];
+export const SECTIONS = {
+  'components': 'Components',
+};
+const CDK = 'cdk';
+const COMPONENTS = 'components';
```
Comment (jelbourn) [inline src/app/shared/documentation-items/documentation-items.ts]: Is it worthwhile to have these as constants?

**Comment by josephperrott** [comment]
@jelbourn This is ready for review once more.

**Pull request merged**